src/dmd/dtemplate.d

```diff
@@ -794,6 +794,31 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
         return result;
     }
 
+    /******************************
+     * Create a scope for the parameters of the TemplateInstance
+     * ti in the parent scope sc from the ScopeDsymbol paramsym.
+     *
+     * If paramsym is null a new ScopeDsymbol is used in place of
+     * paramsym.
+     * Params:
+     *      ti         = the TemplateInstance whose parameters to generate the scope for.
+     *      sc         = the parent scope of ti
+     *      paramsym   = the ScopeDsymbol to create the scope under
+     * Returns:
+     *      a scope for the parameters of ti
+     */
+    Scope* scopeForTemplateParameters(TemplateInstance ti, Scope* sc, ScopeDsymbol paramsym)
```

Comment:
Wouldn't it better to define `paramsym` as an `out` variable then?

Reply:
I can't remember why I did that. I'll take another look tomorrow morning.